### PR TITLE
[Don't squash me!] Rename CSM flavours to CSM restrictions

### LINKED ISSUE #7467: Rename CSM_FLAVOUR_LIMITS
The naming seems a little odd, i think it's better we use a consistent and more descriptive term like 'CSM DISABLE FLAGS' or similar, because it is a byteflag.
'Flavour' needs changing anyway due to not being American English.

### PR DESCRIPTION
**Pull request by SmallJoker**
Also update the translations.
Fixes #7467 and #6690

### CODE REVIEW

File: builtin/settingtypes.txt [hunk 1]
```diff
@@ -1135,13 +1135,12 @@ server_side_occlusion_culling (Server side occlusion culling) bool true
...
-csm_flavour_limits (Client side modding flavour limits) int 18
+#    LOOKUP_NODES_LIMIT: 16 (limits get_node call client-side to csm_restriction_noderange)
+csm_restriction_flags (Client side modding restrictions) int 18
 
-#   If the CSM flavour for node range is enabled, get_node is limited to
-#   this many nodes from the player.
-csm_flavour_noderange_limit (Client side noderange flavour limit) int 8
+#   If the CSM restriction for node range is enabled, get_node calls are limited
+#   to this distance from the player to the node.
+csm_restriction_noderange (Client side node lookup restriction) int 8
```
Comment (paramat) [inline builtin/settingtypes.txt]: Maybe:
Client side node lookup range restriction

**Comment by paramat** [comment]
+1 once line comment is attended to, and if tested.

**Comment by Fixer-007** [comment]
Makes sense, flavours is confusing to nonenglish speaker.

**Comment by nerzhul** [comment]
@Fixer-007 i'm not english and i was the naming owner :)

**Comment by nerzhul** [comment]
:+1: once LINT is okay

**Comment by SmallJoker** [comment]
@nerzhul LINT is asking for [non-rule conformant changes](https://travis-ci.org/minetest/minetest/jobs/395038451). Should I disable it manually for those lines?

**Comment by paramat** [comment]
Please add the file to the whitelist, don't switch linter on and off in the code itself.

**Comment by nerzhul** [comment]
i disagree with the whitelisting for 2 lines. whitelisting is good for a big legacy file, not for a proper file. Would you like to remove the whole linter and loose many time to review the code style of all PR again ? It seems lint has proven that we win some time

**Comment by paramat** [comment]
I support the linter and don't want to remove it. But also don't want code filled with on/off switching of linter checks, because if we do that it will eventually be needed in thousands of places and will make a mess of the code. What i tend to do for a whitelisted file is remove it from the whitelist temporarily to see what is reported and attend to that.

**Comment by paramat** [comment]
The usefulness of the linter is overstated, it does a terrible job of reviewing codestyle, 95+% of what it reports is useless due to inability to cope with our codestyle. It also cannot judge when the code does not follow our own codestyle as a human reviewer can.

Taking into account the work we do to keep it happy and not cause builds to fail, and time scanning through the mostly useless reports i'm not sure it saves any time and it is debateable whether it is worth having. I have rarely found it useful but very often see it being irritating and wasting devs' time.

**Comment by SmallJoker** [comment]
> 95+% of what it reports is useless

For me that's definitely lower than 10%, which includes this PR. Disabling LINT is a bad choice, so a simple workaround should be found to at least keep the code style acceptable for the majority of the code. I'd rather add six (?) comment lines to disable LINT for these lines instead of disabling it for the entire file (or even more).

**Comment by paramat** [comment]
That's unnacceptable as explained and will slowly move us towards a nightmare mess. I'll +1 this if you add the file to the whitelist, but i am -1 if linter switches are added, sorry.

**Comment by nerzhul** [comment]
@SmallJoker i approve you choice + the PR, fix the LINT problem and merge :)

**Comment by paramat** [comment]
I'm outnumbered here, you 2 have the approvals necessary so SmallJoker i suggest doing what you prefer, either way you will get 2 approvals.

It's good to know others find the linter more useful than i do, i probably see the worst of it because it really doesn't like the mapgen files. It's probably worth having, but i am very concerned about where this switching method will lead.

Think about what will happen to a file as this method is applied repeatedly, there will eventually be very many switching lines bloating the code, and managing these lines is more time and effort than my alternative approach described above.

If the linter was a person reviewing code, they would be considered almost useless and annoying, with no knowledge of our own codestyle and reviewing according to a different codestyle. We wouldn't fill files with little comments saying "ignore this bit" :)

**Comment by SmallJoker** [comment]
db42542e271a00efd017b192f3a63e04f5b3dbf4
b589352e79de438590d94f98783dbf1601454d91
Merged.

**Pull request closed without merging**